sycl/include/sycl/detail/common.hpp

```diff
@@ -131,6 +131,8 @@ class __SYCL_EXPORT tls_code_loc_t {
   /// @details If a previous populated TLS entry exists, this constructor will
   /// capture the informationa and allow you to query the information later.
   tls_code_loc_t();
+  tls_code_loc_t(const tls_code_loc_t &TLSCodeLoc) = default;
+  tls_code_loc_t &operator=(const tls_code_loc_t &TLSCodeLoc) = default;
```

Comment:
This class just has the `MLocalScope` member variable, so default works fine here

Reply:
I think this should be `delete` instead. One of the main jobs of `tls_code_loc_t` is to maintain the global state in `GCodeLocTLS`. If we copy object, we could end up with the copy resetting the global state before the top-level copy of it.

Reply:
ok, fixed in the latest commit